### PR TITLE
Added dictionary filter by name

### PR DESCRIPTION
**Pull request by Lantzify**
### Description

Just added a simple `umb-search-filter` in order to better find dictionary items when they start to pile up.

### CODE REVIEW

**Review by mikecp — CHANGES_REQUESTED** [review]
Hi @Lantzify ,

Thanks a lot for this new nice addition to the dictionary section 👍
I just added a few comments regarding localization, if you could have a look at them and update the PR, it would be just great 😁
Cheers!

File: src/Umbraco.Core/EmbeddedResources/Lang/en_us.xml [hunk 1]
```diff
@@ -588,6 +588,7 @@
     <key alias="parentDoesNotExists">Parent item does not exist.</key>
     <key alias="noItems">There are no dictionary items.</key>
     <key alias="noItemsInFile">There are no dictionary items in this file.</key>
+    <key alias="noItemsFound">There were no dictionary items found.</key>
```
[the remark this answers is not in the document]
Comment (Lantzify) [reply src/Umbraco.Core/EmbeddedResources/Lang/en_us.xml:591]: Absolutely! Missed that one 😅

File: src/Umbraco.Web.UI.Client/src/views/dictionary/list.html [hunk 2]
```diff
@@ -23,6 +23,13 @@
               </umb-button>
             </umb-editor-sub-header-content-left>
 
+            <umb-editor-sub-header-content-right>
+              <umb-search-filter model="vm.filter.searchTerm"
+                                 label-key="placeholders_filter"
+                                 text="Type to filter...">
```
[the remark this answers is not in the document]
Comment (Lantzify) [reply src/Umbraco.Web.UI.Client/src/views/dictionary/list.html:29]: Isn't it already localized? I changed the backoffice langue to Swedish. 🤔
Capture

[the remark this answers is not in the document]
Comment (mikecp) [reply src/Umbraco.Web.UI.Client/src/views/dictionary/list.html:29]: Indeed, I did not pay attention to the `label-key` attribute 😅

**Comment by mikecp** [comment]
Hi @Lantzify ,

Thanks for the update, I've tested and all works fine 👍🎉
I just have a last concern about the "order by name" part after the filter. In the current situation, the items in the table are displayed in the same order as in the dictionary structure:

image

With the "order by name", they are displayed alphabetically as expected :

image

I think this might be confusing at first load and when the filter is left empty, because then it's kind of a change in behavior from previously (and I'm afraid it might even be seen as breaking change by HQ team), but I am also wondering whether having the ordering by name when the filter is filled in has any specific added value compared to keeping the items in original order.
What's your opinion about it? Maybe I am overseeing something?

Cheers!

**Review by mikecp — APPROVED** [review]

**Comment by mikecp** [comment]
Hi @Lantzify ,

Thank you for the update, I think it is good to go and get merged now 👍😅
But it would be great indeed to have sort action to the dictionary, I guess this makes a new topic of PR for you 😁😉

Thanks again for all the love you put in the dictionary section, I think it has improved a lot in the last few months thanks to all your PR's 👍👍

Cheers and H5YR !!

**Pull request merged**